utils/measure/measure/powermeter/shelly.py

```diff
@@ -121,5 +130,12 @@ def _detect_api_version(self) -> int:
         _LOGGER.debug("Shelly API version %d detected", gen)
         return int(gen)
 
+    def has_voltage_support(self) -> bool:
+        if isinstance(self.api, ShellyApiGen1):
+            return False
+        if isinstance(self.api, ShellyApiGen2Plus):
+            return True
+        return False
```

Comment:
Only Shelly gen 2 supports voltage so we can simplify the logic.

```suggestion
        return isinstance(self.api, ShellyApiGen2Plus)
```

Reply:
yeah, I wanted to keep this open, in case they release a gen3 or gen4 or so. But I agree, that's fine in the meantime :)